### PR TITLE
Misc fixes for s_client with QUIC

### PR DESCRIPTION
**Pull request by mattcaswell**
This same out of trying to make s_client try to play nicely with the quicserver utility (util/quicserver.c).

It turns out that the s_client QUIC functionality was broken sometime after it was implemented by changes in the QUIC code. We need to find a reasonable way of testing s_client in QUIC mode in the test framework (we can't do this at the moment).

I also fixed one error in quicserver which I found when connecting to it from s_client.

### CODE REVIEW

File: ssl/quic/quic_impl.c [hunk 1]
```diff
@@ -2287,9 +2289,15 @@ static size_t ossl_quic_pending_int(const SSL *s, int check_channel)
     size_t avail = 0;
     int fin = 0;
 
-    if (!expect_quic_with_stream_lock(s, /*remote_init=*/-1, &ctx))
+
```
Comment (hlandau) [inline ssl/quic/quic_impl.c:2292]: Can you explain the rationale behind this change?

Comment (mattcaswell) [reply ssl/quic/quic_impl.c:2292]: s_client calls `SSL_has_pending()` very early - even before a stream has been associated with the SSL object (during handshake). The TLS implementation just returns 0 in this case (i.e. no data pending), but `expect_quic_with_stream_lock` fails if there is no stream raising a non-normal error and causing subsequent calls to `SSL_get_error()` to return `SSL_ERROR_SSL`.

Comment (hlandau) [reply ssl/quic/quic_impl.c:2292]: OK, makes sense.

Comment (t8m) [reply ssl/quic/quic_impl.c:2292]: Hmm... how does SSL_read()/SSL_write() work in this regard?

Comment (t8m) [reply ssl/quic/quic_impl.c:2292]: Looking at the source it seems SSL_read() should work however SSL_write() doesn't as ossl_quic_write() calls expect_quic_with_stream_lock() at the beginning.

**Review by t8m — APPROVED** [review]

**Review by hlandau — APPROVED** [review]

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**